### PR TITLE
feat: implement per-repository autoPush configuration

### PR DESCRIPTION
**Pull request by adalton**
Replace global autoPushOnComplete setting with per-repository autoPush field to enable granular control of git push behavior for agentic sessions. Each repository can now be configured independently with autoPush: true/false.

## Changes

- Add autoPush field to SessionRepo type (backend, frontend, CRD)
- Remove deprecated autoPushOnComplete from all components
- Add type validation in runner to handle invalid autoPush values
- Generate Git Push Instructions in system prompt for repos with autoPush=true
- Add comprehensive test coverage (10 backend tests, 19 runner tests)
- Update UI with clear help text explaining autoPush behavior

## Testing

- All 351 backend tests pass
- All 19 runner tests pass (including edge case type validation tests)
- Frontend builds with 0 errors, 0 warnings
- Manual testing verified correct behavior

### CODE REVIEW

File: components/runners/claude-code-runner/adapter.py [hunk 1]
```diff
@@ -1287,6 +1306,20 @@ def _build_workspace_context_prompt(self, repos_cfg, workflow_name, artifacts_pa
...
+                prompt += "## Git Push Instructions\n\n"
+                prompt += "The following repositories have auto-push enabled. When you make changes to these repositories, you MUST commit and push your changes:\n\n"
+                for repo in auto_push_repos:
+                    repo_name = repo.get('name', 'unknown')
+                    repo_branch = repo.get('branch', 'main')
+                    prompt += f"- **repos/{repo_name}/** (branch: {repo_branch})\n"
+                prompt += "\nAfter making changes to any auto-push repository:\n"
+                prompt += "1. Use `git add` to stage your changes\n"
+                prompt += "2. Use `git commit -m \"description\"` to commit with a descriptive message\n"
+                prompt += "3. Use `git push origin <branch>` to push to the remote repository\n\n"
```
Comment (Gkrumbach07) [inline components/runners/claude-code-runner/adapter.py:1321]: for context this is what cladue code web says

```
## Git Development Branch Requirements

You are working on the following feature branches:

 **Gkrumbach07/Chores**: Develop on branch `claude/system-prompt-docs-01Eahw2gNvVU8ffDpcM48uea`

### Important Instructions:

1. **DEVELOP** all your changes on the designated branch above
2. **COMMIT** your work with clear, descriptive commit messages
3. **PUSH** to the specified branch when your changes are complete
4. **CREATE** the branch locally if it doesn't exist yet
5. **NEVER** push to a different branch without explicit permission

Remember: All development and final pushes should go to the branches specified above.

## Git Operations

Follow these practices for git:

**For git push:**
- Always use git push -u origin <branch-name>
- CRITICAL: the branch should start with 'claude/' and end with matching session id, otherwise push will fail with 403 http code.
- Only if push fails due to network errors retry up to 4 times with exponential backoff (2s, 4s, 8s, 16s)
- Example retry logic: try push, wait 2s if failed, try again, wait 4s if failed, try again, etc.

**For git fetch/pull:**
- Prefer fetching specific branches: git fetch origin <branch-name>
- If network failures occur, retry up to 4 times with exponential backoff (2s, 4s, 8s, 16s)
- For pulls use: git pull origin <branch-name>

The GitHub CLI (`gh`) is not available in this environment. For GitHub issues ask the user to provide the necessary information directly.


gitStatus: This is the git status at the start of the conversation. Note that this status is a snapshot in time, and will not update during the conversation.
Current branch: claude/system-prompt-docs-01Eahw2gNvVU8ffDpcM48uea

Main branch (you will usually use this for PRs): 

Status:
(clean)

Recent commits:
4bef2c1 Working on making the button turn and stay green
b267637 fixed chore detail view table view color strips and started on making the strip turn green when pressed.
ae1a937 First commit
6f820b8 Initial Commit
```

we can iterate on this prompt. for now it looks good
